server/tests/db/test_01_es5.js

```diff
@@ -0,0 +1,1739 @@
+
+  // Adding numeric properties >= length does increase length:
+  for (var i = 0; i < 5; i++) {
+    a[i] = i;
+    checkLen(i + 1, 'after setting a[' + i + ']');
+  }
+
+  // .length works propery even for large, sparse arrays, and even
+  // if values are undefined:
+  for (i = 3; i <= 31; i++) {
+    var idx = (1 << i) >>> 0;  // >>> 0 converts int32 to uint32
+    a[idx] = undefined;
+    checkLen(idx + 1, 'after setting a[' + idx + ']');
+  }
+
+  // Adding numeric properties < length does not increase length:
+  a[idx - 1] = 'not the largest';
+  checkLen(idx + 1, 'after setting non-largest element');
+
+  // Verify behaviour around largest possible index:
+  a[0xfffffffd] = null;
+  checkLen(0xfffffffe);
+  a[0xfffffffe] = null;
+  checkLen(0xffffffff);
+  a[0xffffffff] = null;  // Not an index.
+  checkLen(0xffffffff);  // Unchanged.
+  a[0x100000000] = null; // Not an index.
+  checkLen(0xffffffff);  // Unchanged.
+
+  function checkIdx(idx, exp, desc) {
+    var r = a.hasOwnProperty(idx);
+    var msg = 'a.hasOwnProperty(' + idx + ') === ' + r;
+    console.assert(r === exp, desc ? msg + ' ' + desc : msg);
+  }
+
+  // Setting length to existing value should have no effect:
+  a.length = 0xffffffff;
+  checkIdx(0xfffffffd, true);
+  checkIdx(0xfffffffe, true);
+  checkIdx(0xffffffff, true);
+  checkIdx(0x100000000, true);
+
+  // Setting length one less than maximum should remove largest
+  // index, but leave properties with keys too large to be indexes:
+  a.length = 0xfffffffe;
+  checkIdx(0xfffffffd, true);
+  checkIdx(0xfffffffe, false);
+  checkIdx(0xffffffff, true);
+  checkIdx(0x100000000, true);
+
+  // Setting length to zero should remove all index properties:
+  a.length = 0;
+  for (var key in a) {
+    if (!a.hasOwnProperty(key)) {
+      continue;
+    }
+    console.assert(String(key >>> 0) !== key || (key >>> 0) === 0xffffffff,
+                   'Setting a.length = 0 failed to remove property ' + key);
+  }
+
+  // Make sure we didn't wipe everything!
+  console.assert(Object.getOwnPropertyNames(a).length === 4,
+                 'Setting .length = 0 removed some non-index properties');
+};
+
+tests.arrayLengthWithNonWritableProps = function() {
+  var a = [];
+  Object.defineProperty(a, 0,
+      {value: 'hi', writable: false, configurable: true});
+  a.length = 0;
+  console.assert(a[0] === undefined && a.length === 0,
+                 'arrayLengthWithNonWritableProps');
+};
+
+tests.arrayLengthWithNonConfigurableProps = function() {
+  var a = [];
+  Object.defineProperty(a, 0,
+      {value: 'hi', writable: false, configurable: false});
+  a.length = 0;
+  console.assert(a[0] === 'hi' && a.length === 1,
+                 'arrayLengthWithNonConfigurableProps');
+};
+
+tests.compValEmptyBlock = function() {
+  console.assert(eval('{}') === undefined, 'compValEmptyBlock');
+};
+
+tests.undefined = function() {
+  console.assert(undefined === undefined, 'undefined');
```

Comment:
I'm not really sure this test is testing anything useful, whereas the original was.  Maybe use `void` to obtain a canonical `undefined` value to compare with the value of the `undefined` variable?